GitVersionCore/BranchNameComparer.cs

```diff
@@ -0,0 +1,18 @@
+namespace GitVersion
+{
+    using System.Collections.Generic;
+    using LibGit2Sharp;
+
+    class BranchNameComparer : IEqualityComparer<Branch>
+    {
+        public bool Equals(Branch x, Branch y)
+        {
+            return x.Name == y.Name;
```

Comment:
hrrm, don't need case insensitivity. Should we disable it for this method? Possible we don't even need this if Branch supports equals and gethashcode properly. I didn't check. @nulltoken ? :)